File: 이준영/백준/Gold/1806. 부분합/부분합.py
# 투 포인터로 해결하였다. (각 원소들이 양수이기에 가능)
# 다 합한 것이 S보다 크거나 같다면 그 때 mn을 최신화하고 left를 1증가시킨다.
# S보다 작다면 right를 1늘린다.
import sys

input = sys.stdin.readline

N, S = map(int, input().split())
arr = list(map(int, input().split()))
mn = 100000

left = 0
right = 0
tot = arr[0]

while True:
    if tot >= S:
        mn = min(mn, right - left + 1)
        tot = tot - arr[left]
        left += 1
    else:
        right += 1
        if right >= N:
            break
        tot += arr[right]

if mn == 100000:
    print(0)
else:
    print(mn)

File: 송인재/백준/Gold/1806. 부분합/부분합.java
import java.io.*;
import java.util.*;

public class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine());
    int n = Integer.parseInt(st.nextToken());
    int s = Integer.parseInt(st.nextToken());
    int[] nums = new int[n + 1];
    st = new StringTokenizer(br.readLine());
    int min = 100_000;

    for (int i = 1; i <= n; i++) {
      int num = Integer.parseInt(st.nextToken());
      nums[i] = nums[i - 1] + num;
    }

    if (nums[n] < s) {
      System.out.println(0);

      return;
    }

    int left = 0;
    int right = 0;

    while (right <= n) {
      if (nums[right] < nums[left] + s) {
        right++;
      } else {
        min = Math.min(min, right - left++);
      }
    }

    System.out.println(min);
  }

}
